### DISCUSSION TITLE
Query with panic error: expected 0 arguments, got 1

### DISCUSSION THREAD

**Discussion by scarlosantos** [godror/godror#207] [Q&A] [answered]
Hi,

This code has raised panic error, with message "sql: expected 0 arguments, got 1" when executed:

```
row:= db.QueryRow(`SELECT DUMMY FROM DUAL WHERE 1=?`, 1)
err := row.Scan(&test)

if err != nil {
    panic(err)
}
```
go version go1.17.5 darwin/amd64

godror version: v0.30.0

Any suggestions on how to fix this?

Best regards,

**Comment by tgulacsi** [comment, marked as answer]
Oracle uses :1, :2, ... as placeholder, not ?.
________________________________
From: Sebastião Carlos Santos ***@***.***>
Sent: Sunday, January 2, 2022 7:54:07 PM
To: godror/godror ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [godror/godror] Query with panic error: expected 0 arguments, got 1 (Discussion #207)


Hi,

This code has raised panic error, with message "sql: expected 0 arguments, got 1" when executed:

row:= db.QueryRow(`SELECT DUMMY FROM DUAL WHERE 1=?`, 1)
err := row.Scan(&test)

if err != nil {
    panic(err)
}


go version go1.17.5 darwin/amd64

godror version: v0.30.0

Any suggestions on how to fix this?

Best regards,

—
Reply to this email directly, view it on GitHub<https://github.com/godror/godror/discussions/207>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AAB6QSQ3CGGEXXWIG2ICJFLUUCNM7ANCNFSM5LD6PTUA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by scarlosantos** [reply]
Thanks.... Works fine....
Best regards.
